engine/src/main/scala/cromwell/webservice/routes/wes/CromwellMetadata.scala

```diff
@@ -0,0 +1,101 @@
+package cromwell.webservice.routes.wes
+
+import spray.json.{JsObject, JsonFormat, JsonParser}
+
+final case class CromwellSubmittedFiles(workflow: Option[String],
+                                        workflowType: Option[String],
+                                        workflowTypeVersion: Option[String],
+                                        options: Option[String],
+                                        inputs: Option[String],
+                                        labels: Option[String]
+                                       )
+
+final case class CromwellCallsMetadata(shardIndex: Option[Int],
+                                       commandLine: Option[String],
+                                       returnCode: Option[Int],
+                                       start: Option[String],
+                                       end: Option[String],
+                                       stdout: Option[String],
+                                       stderr: Option[String]
+                                      )
+
+final case class CromwellMetadata(workflowName: Option[String],
+                                  id: String,
+                                  status: String,
+                                  start: Option[String],
+                                  end: Option[String],
+                                  submittedFiles: CromwellSubmittedFiles,
+                                  outputs: Option[JsObject],
+                                  calls: Option[Map[String, Seq[CromwellCallsMetadata]]]
+                                 ) {
+  import CromwellMetadata._
+
+  def wesRunLog: WesRunLog = {
+    val workflowParams = submittedFiles.inputs.map(JsonParser(_).asJsObject)
+    val workflowTags = submittedFiles.labels.map(JsonParser(_).asJsObject)
+    val workflowEngineParams = submittedFiles.options.map(JsonParser(_).asJsObject)
+
+    val workflowRequest = WesRunRequest(workflow_params = workflowParams,
+      workflow_type = submittedFiles.workflowType.getOrElse("Unable to find workflow type"),
+      workflow_type_version = submittedFiles.workflowTypeVersion.getOrElse("Unable to find workflow version"),
```

Comment:
If a user doesn't submit these values, I think it's strange for the metadata to respond with "unable to find". I'd probably just leave them as empty strings (or maybe "not supplied"?)... or does Cromwell normally fill them in even if the request didn't include them?

Reply:
I decided to change this to "not supplied," as the POST endpoint allows for these values to be input, and I think it makes it more obvious to a user what is going on here.